src/Entities/BaseEntity.php

```diff
@@ -57,8 +57,13 @@ protected function map(\SimpleXMLElement $xml): Collection
         return $collection;
     }
 
-    protected function call(DataType $dataType, string $endpoint, string $identifier = null, array $attributes = []): ?\SimpleXMLElement
+    /** @param string|int|null $identifier */
+    protected function call(DataType $dataType, string $endpoint, $identifier = null, array $attributes = []): ?\SimpleXMLElement
```

Comment:
$identifier should always be a string except for index calls where it can be null.
I think the issue lies with the BaseResource's getId method. This should always return a string, but might not in this case.
I suggest enforcing the type on the identifier either in the save method or the getId method itself.

Reply:
I have opted to fix this in the save method to maintain backwards compatibility.